Feature/add my page

Yang-Seungmin opened the pull request:
### 마이페이지 화면 추가
현재 포인트 수치 및 활동 개수를 파악할 수 있으며 설정, 내 스크랩, 구입한 파일 확인 가능
_추후 설정 화면이 만들어지면 설정 버튼 클릭 리스너 추가_

### 내 스크랩
유저가 스크랩한 강의 목록을 확인할 수 있으며 삭제할 수 있는 기능 제공
_강의 평가 화면이 만들어지면 강의 클릭 리스너 추가_

### 구입한 자료
유저가 구입한 강의자료 목록을 확인할 수 있다
_강의자료 상세페이지 화면이 만들어지면 강의자료 목록 클릭 리스너 추가 _

Yang-Seungmin committed:
Added mypage layout

Yang-Seungmin committed:
Added user point api

Yang-Seungmin committed:
Added point record activity layout

Yang-Seungmin committed:
Added mypage api

Yang-Seungmin committed:
Added purchased file list

Yang-Seungmin committed:
Added purchased bank activity

Yang-Seungmin committed:
Added scrap lectures activity

Yang-Seungmin committed:
Fixed my scrap recyclerview selection problem

Yang-Seungmin committed:
Fixed my scrap lecture cannot remove item

Yang-Seungmin committed:
Fixed that my page edit mode switch problem

Yang-Seungmin committed:
Code cleanup

yunjaena commented on app/src/main/java/in/hangang/hangang/api/AuthApi.kt:
```diff
@@ -14,4 +17,26 @@ interface AuthApi {
+    @POST(LECTURE_SCRAPED)
+    fun addScrapLecture(@Body scrapLectureRequest: ScrapLectureRequest): Single<CommonResponse>
+
+    @HTTP(method = "DELETE", path = LECTURE_SCRAPED, hasBody = true)
```
@DELETE annotation을 사용하지 않은 이유가 궁금합니다

Yang-Seungmin replied:
Http Body 가 포함된 DELETE API에서 
`java.lang.IllegalArgumentException: Non-body HTTP method cannot contain @Body.`
오류가 발생하여 이렇게 해결했습니다

Yang-Seungmin committed:
Modified scrap lecture removal logic

Yang-Seungmin committed:
Rename file

Yang-Seungmin committed:
Fixed some errors

- Point record recyclerview date format and listing(flip list order)
- my page fragment horizontal divider not showing
- purchased bank recyclerview orientation

Yang-Seungmin committed:
Merge branch 'develop' into feature/add_my_page

# Conflicts:
#	app/build.gradle
#	app/src/main/AndroidManifest.xml
#	app/src/main/java/in/hangang/hangang/api/AuthApi.kt
#	app/src/main/java/in/hangang/hangang/constant/ApiConstant.kt
#	app/src/main/java/in/hangang/hangang/data/entity/HashTag.kt
#	app/src/main/java/in/hangang/hangang/data/entity/Lecture.kt
#	app/src/main/java/in/hangang/hangang/data/request/EmailConfigRequest.kt
#	app/src/main/java/in/hangang/hangang/data/request/EmailRequest.kt
#	app/src/main/java/in/hangang/hangang/data/source/LectureDataSource.kt
#	app/src/main/java/in/hangang/hangang/data/source/LectureRepository.kt
#	app/src/main/java/in/hangang/hangang/data/source/local/LectureLocalDataSource.kt
#	app/src/main/java/in/hangang/hangang/data/source/remote/LectureRemoteDataSource.kt
#	app/src/main/java/in/hangang/hangang/data/source/remote/UserRemoteDataSource.kt
#	app/src/main/java/in/hangang/hangang/data/source/repository/UserRepository.kt
#	app/src/main/java/in/hangang/hangang/di/DataSourceModule.kt
#	app/src/main/java/in/hangang/hangang/di/RepositoryModule.kt
#	app/src/main/java/in/hangang/hangang/di/ViewModelModule.kt
#	app/src/main/java/in/hangang/hangang/ui/changepassword/viewmodel/ChangePasswordFragmentViewModel.kt
#	app/src/main/java/in/hangang/hangang/ui/dashboard/DashBoardViewModel.kt
#	app/src/main/java/in/hangang/hangang/ui/home/findbymajor/adapter/FindByMajorRecyclerViewAdapter.kt
#	app/src/main/java/in/hangang/hangang/ui/home/ranking/adapter/RankingListAdapter.kt
#	app/src/main/java/in/hangang/hangang/ui/home/ranking/fragment/RankingListFragment.kt
#	app/src/main/java/in/hangang/hangang/ui/home/recentlectures/adapter/RecentLecturesRecyclerViewAdapter.kt
#	app/src/main/java/in/hangang/hangang/ui/login/LoginActivity.kt
#	app/src/main/java/in/hangang/hangang/ui/login/LoginViewModel.kt
#	app/src/main/java/in/hangang/hangang/ui/main/MainActivity.kt
#	app/src/main/java/in/hangang/hangang/ui/signup/activity/SignUpEmailActivity.kt
#	app/src/main/java/in/hangang/hangang/ui/signup/activity/SignUpMajorActivity.kt
#	app/src/main/java/in/hangang/hangang/ui/signup/viewmodel/SignUpMajorViewModel.kt
#	app/src/main/res/drawable/ic_chevron_right.xml
#	app/src/main/res/layout/fragment_dashboard.xml
#	app/src/main/res/navigation/mobile_navigation.xml
#	app/src/main/res/values/strings.xml

Yang-Seungmin committed:
Moved files

Yang-Seungmin committed:
Removed duplicated files

Yang-Seungmin committed:
Added lecture bank my scrap activity

Yang-Seungmin committed:
Added lecture bank my scrap activity

Yang-Seungmin committed:
Removed duplicated user

Yang-Seungmin committed:
Fixed file structure

Yang-Seungmin committed:
Merge branch 'develop' into feature/add_my_page

Yang-Seungmin committed:
Merge branch 'develop' into feature/add_my_page

Yang-Seungmin committed:
Fixed some errors
